katdal/chunkstore_s3.py

```diff
@@ -154,12 +154,10 @@ def _read_chunk(response):
     """Efficiently read NumPy array in NPY format from content of HTTP response."""
     data = response.raw
     # Workaround for https://github.com/urllib3/urllib3/issues/1540
-    # On Python 2, http.client.HTTPResponse doesn't implement readinto.
     # We also can't use the workaround if the content is encoded (e.g.
     # gzip compressed) because that's decoded in urllib3, not httplib.
     if ('Content-encoding' not in response.headers
-            and hasattr(data, '_fp')
-            and hasattr(data._fp, 'readinto')):
+            and hasattr(data, '_fp')):
```

Comment:
I think it might be worth leaving this, just in case there are conditions under which HTTPResponse ._fp doesn't provide a readinto (since it is a private member there is no guarantee of anything).

Reply:
I wondered about that too... There is also `support_chunked_reads()` which is a public function which implicitly checks for `_fp`, but the same issues arise. I'll put it back.